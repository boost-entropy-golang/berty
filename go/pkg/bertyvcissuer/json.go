package bertyvcissuer

type JSONError struct {
	Error string `json:"error"`
}

type JSONChallenge struct {
	Challenge string `json:"challenge"`
}
