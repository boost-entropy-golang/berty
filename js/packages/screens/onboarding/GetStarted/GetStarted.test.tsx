import { renderScreen } from '@berty/utils/testing/renderScreen.test'
import * as testIDs from '@berty/utils/testing/testIDs.json'

import { GetStarted } from './GetStarted'

test('Onboarding.GetStarted renders correctly', async () => {
	const { toJSON } = renderScreen('Onboarding.GetStarted', GetStarted)
	expect(toJSON()).toMatchSnapshot()
})

/**
 * This test is only there as an example of getting an element by testID,
 * it shouldn't be needed due to the snapshot test
 */
test('create button exists', async () => {
	const { getByTestId } = renderScreen('Onboarding.GetStarted', GetStarted)

	const createButton = getByTestId(testIDs['create-account-button'])
	expect(createButton).toBeTruthy()
})
