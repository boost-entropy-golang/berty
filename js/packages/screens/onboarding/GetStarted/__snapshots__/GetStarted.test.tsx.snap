// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding.GetStarted renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 16,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 32,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <SvgMock />
  </View>
  <View>
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 30,
            },
            Object {
              "color": "#383B62",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 40,
              },
              Object {
                "fontFamily": "Bold Open Sans",
              },
              Object {
                "color": "#525BEC",
                "textTransform": "uppercase",
              },
            ],
          ]
        }
      >
        onboarding.getstarted.title
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 9,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 30,
            },
            Object {
              "color": "#383B62",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "textAlignVertical": "bottom",
              },
              Object {
                "fontFamily": "Italic Open Sans",
              },
            ],
          ]
        }
      >
        onboarding.getstarted.desc
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 32,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#525BEC",
            "borderRadius": 9,
            "marginTop": 9,
            "opacity": 1,
            "padding": 16,
            "width": 250,
          }
        }
      >
        <Text
          accessibilityLabel="onboarding.getstarted.create-button"
          style={
            Array [
              Object {
                "fontSize": 30,
              },
              Object {
                "color": "#383B62",
                "fontFamily": "Open Sans",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "Bold Open Sans",
                },
                Object {
                  "color": "#FFFFFF",
                  "textTransform": "uppercase",
                },
                undefined,
              ],
            ]
          }
        >
          onboarding.getstarted.create-button
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#525BEC20",
            "borderRadius": 9,
            "marginTop": 9,
            "opacity": 1,
            "padding": 16,
            "width": 250,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
              },
              Object {
                "color": "#383B62",
                "fontFamily": "Open Sans",
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "Bold Open Sans",
                },
                Object {
                  "color": "#525BEC",
                  "textTransform": "uppercase",
                },
                undefined,
              ],
            ]
          }
        >
          onboarding.getstarted.import-button
        </Text>
      </View>
    </View>
  </View>
</View>
`;
